### ISSUE TITLE
RNNoise integration — open-source neural network noise suppression

### ISSUE THREAD

**Issue by ten9876** [ten9876/AetherSDR#82]
## Feature

Integrate [RNNoise](https://github.com/xiph/rnnoise) (Mozilla/Xiph.Org) as an additional client-side noise suppression option alongside NR2.

## Why

RNNoise is a recurrent neural network trained specifically for voice noise suppression. It's:
- **BSD-3 licensed** — compatible with GPLv3
- **~100KB** total footprint (model + code) — no GPU required
- **Real-time** — designed for low-latency audio processing
- **Proven** — used by PipeWire, NoiseTorch, WebRTC, Discord
- **Platform-agnostic** — pure C, runs on any CPU with optional AVX2/SSE4.1

Compared to NR2 (Ephraim-Malah spectral subtraction), RNNoise uses a deep learning model trained on thousands of hours of noisy speech. It's particularly better at preserving voice naturalness while removing non-stationary noise.

## API

```c
DenoiseState *st = rnnoise_create(NULL);  // NULL = built-in model
float frame[480];  // 48kHz, 10ms frames
float vad_prob = rnnoise_process_frame(st, frame, frame);  // in-place
rnnoise_destroy(st);
```

## Technical Constraints

| Parameter | Value |
|-----------|-------|
| Sample rate | **48kHz** (mandatory) |
| Frame size | **480 samples** (10ms at 48kHz) |
| Channels | **Mono** only |
| Format | **float** (-32768 to 32768 range, NOT -1.0 to 1.0) |
| Latency | ~10ms |

### Phase 1: Core Integration

**New file: `src/core/RNNoiseFilter.h/.cpp`**
- Wraps the RNNoise C API in a C++ class
- Handles sample rate conversion: our audio is 24kHz, RNNoise needs 48kHz
  - Upsample 24k→48k (duplicate samples or linear interpolation)
  - Process through RNNoise at 48kHz
  - Downsample 48k→24k
- Frame buffering: accumulate 480 samples (10ms at 48kHz) before processing
- Mono conversion: average L+R → process → duplicate to stereo

### Phase 2: AudioEngine Integration

**Modify `AudioEngine::feedAudioData()`:**
- Add RNNoise as a third noise reduction option alongside NR (radio) and NR2 (spectral)
- Process chain: raw PCM → RNNoise → output (or NR2 → RNNoise for stacking)
- Enable/disable via signal from GUI

### Phase 3: GUI

**NR button 4-state cycle:**
Off → NR → NR2 → RNN-AI → Off

Or separate toggle in DSP panel labeled "AI" or "RNN-AI" to distinguish from the radio's RNN.

### Phase 4: Build System

**CMakeLists.txt:**
- Optional dependency: `pkg_check_modules(RNNOISE rnnoise)`
- Fallback: bundle RNNoise source directly (it's small — ~20 source files)
- `#ifdef HAVE_RNNOISE` guards
- AppImage/Windows/macOS: bundle the library

### Sample Rate Considerations

RNNoise requires 48kHz. Our audio pipeline is 24kHz. Two approaches:

**Option A: Upsample → process → downsample**
- Simple 2x interpolation (same as our existing 48kHz fallback)
- Slight quality loss from resampling
- No changes to AudioEngine's core pipeline

**Option B: Process at 48kHz throughout**
- Change AudioEngine to work at 48kHz internally
- Only downsample for the radio's VITA-49 TX path
- Better quality but larger refactor
- Aligns with the 48kHz fallback we already added

Option A for initial implementation, Option B as follow-up.

### vs. Existing NR2

| Feature | NR2 (Ephraim-Malah) | RNNoise |
|---------|---------------------|---------|
| Algorithm | Spectral subtraction + OSMS | Recurrent neural network |
| Training | None (analytical) | Pre-trained on speech + noise |
| Noise types | Stationary noise | All types (voice-optimized) |
| Musical noise | Possible (mitigated by smoothing) | Rare (learned suppression) |
| CPU cost | ~1-2% one core | ~2-3% one core |
| Dependency | FFTW3 (optional) | librnnoise (~100KB) |
| License | Part of AetherSDR (GPLv3) | BSD-3 |

Both can potentially be stacked for maximum noise reduction.

## References
- RNNoise: https://github.com/xiph/rnnoise
- Paper: https://arxiv.org/abs/1709.08243 (Valin, 2018)
- Demo: https://jmvalin.ca/demo/rnnoise/
